tests/integration/nm/vlan_test.py

```diff
@@ -67,9 +67,10 @@ def _create_vlan(vlan_desired_state):
     with mainloop():
         con_profile = nm.connection.create_profile((con_setting, vlan_setting))
         nm.connection.add_profile(con_profile, save_to_disk=False)
-    with mainloop():
-        remote_profile = nm.nmclient.client().get_connection_by_id(ifname)
-        nm.device.activate(dev=None, connection=remote_profile)
+        nm.device.activate(
+            dev=None,
+            fetch_remote_connection_func=(
+                lambda: nm.nmclient.client().get_connection_by_id(ifname)))
```

Comment:
It seems that `activate()` could be easily be simplified by just accepting a connection id instead of the fetch function.

Reply:
see previous answer to this